gridstatus/spp.py

```diff
@@ -193,6 +201,189 @@ def get_load_forecast(self, date, forecast_type="MID_TERM", verbose=False):
+        # ensure there's data for the complete day
+        if date.date() < now.date() and not end:
+            date = date.normalize() + pd.Timedelta(days=1, minutes=5)
+
+        df = self._get_5_min_solar_and_wind_forecast_and_actuals(
+            date.floor("5T"),
+            end=end or end_override,
+            verbose=verbose,
+        )
+
+        df.columns = [col.strip() for col in df.columns]
+
+        # Drop duplicated intervals, keeping the most recent Publish Time values.
+        # The actual values should be the same at each interval
+        df = df.drop_duplicates(subset=["Interval Start", "Interval End"], keep="last")
+
+        # Don't include the forecast to avoid confusion
+        cols_to_drop = [col for col in df if "Forecast" in col]
+
+        df = df.drop(columns=cols_to_drop + ["Publish Time"]).dropna(
+            subset=["Actual Wind MW", "Actual Solar MW"],
+        )
+
+        if end:
+            return df[df["Interval Start"].between(date, end)]
+
+        return df
+
+    @support_date_range("HOUR_START")
+    def _get_hourly_solar_and_wind_forecast(self, date, end=None, verbose=False):
+        """System-wide wind and solar forecast data for +7days by hour."""
+        url = self._midterm_solar_and_wind_url_from_date(date)
+
+        df = pd.read_csv(url)
+        df = self._process_solar_and_wind_load_or_forecast(
+            df,
+            url,
+            end_time_col="GMTIntervalEnd",
+            interval_duration=pd.Timedelta(hours=1),
+        )
+
+        return df
+
+    @support_date_range("5_MIN")
+    def _get_5_min_solar_and_wind_forecast_and_actuals(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """System-wide wind and solar forecast data for +4 hours in 5-minute intervals.
+        Including actual wind and solar resource totals from previous intervals.
+        """
+        url = self._shortterm_solar_and_wind_url_from_date(date)
+
+        df = pd.read_csv(url)
+        # According to the docs, the end time col should be GMTIntervalEnd, but it's
+        # only GMTInterval in the data
+        df = self._process_solar_and_wind_load_or_forecast(
+            df,
+            url,
+            end_time_col="GMTInterval",
+            interval_duration=pd.Timedelta(minutes=5),
+        )
+
+        return df
+
+    def _process_solar_and_wind_load_or_forecast(
+        self,
+        df,
+        url,
+        end_time_col,
+        interval_duration,
+    ):
+        df = self._handle_market_end_to_interval(
+            df,
+            end_time_col,
+            interval_duration,
+        )
+
+        # Publish time is only an estimate (we'd have to scrape the page to get the
+        # actual publish time)
+        # Generally accurate to within an hour for midterm and 5 mins for shortterm
+        df["Publish Time"] = pd.Timestamp(url.split("-")[-1].split(".")[0])
```

Comment:
i would consider the time in the file to be the official "publish time" in the case of regular forecast

Reply:
The time in the filename? The times listed on the webpage aren't the same as the time in the file name, but that could just be the time they are posted. 

<img width="892" alt="Screenshot 2024-01-27 at 20 19 36" src="https://github.com/kmax12/gridstatus/assets/17484616/02782b23-9932-4420-a9c0-d1cb11485483">